src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs

```diff
@@ -824,8 +824,13 @@ private static void ValidateUserDefinedConditionalLogicOperator(ExpressionType n
             {
                 left = left.GetNonNullableType();
             }
-            MethodInfo opTrue = TypeUtils.GetBooleanOperator(method.DeclaringType, "op_True");
-            MethodInfo opFalse = TypeUtils.GetBooleanOperator(method.DeclaringType, "op_False");
+            Type declaringType = method.DeclaringType;
+            if (declaringType == null)
```

Comment:
same thing about  Type and "=="

Actually, It seems MethodInfo also has "==" implemented. 
Perhaps it is worth going through and turn trivial null checks with Type/MemberInfo and the like into object equality or ?. operators. As a separate change of course.